Source/player.cpp

```diff
@@ -1683,152 +1683,109 @@ void __fastcall RespawnDeadItem(ItemStruct *itm, int x, int y)
-	v20 = (unsigned char)gbMaxPlayers > 1u && plr[v3 / 0x54D8].plrlevel == 16;
-	if ( v2 >= 4 )
-		TermMsg("StartPlayerKill: illegal player %d", v2);
-	v4 = plr[v3 / 0x54D8]._pClass;
-	if ( v4 )
-	{
-		if ( v4 == 1 )
-		{
-			v5 = PS_ROGUE71;
-		}
-		else
-		{
-			if ( v4 != 2 )
-				goto LABEL_18;
-			v5 = PS_MAGE71;
-		}
-		PlaySfxLoc(v5, plr[v3 / 0x54D8].WorldX, plr[v3 / 0x54D8].WorldY);
-		goto LABEL_18;
 	}
-	PlaySfxLoc(PS_DEAD, plr[v3 / 0x54D8].WorldX, plr[v3 / 0x54D8].WorldY); /// BUGFIX: should use `PS_WARR71` like other classes
-LABEL_18:
-	if ( plr[v3 / 0x54D8]._pgfxnum )
-	{
-		plr[v3 / 0x54D8]._pgfxnum = 0;
-		plr[v3 / 0x54D8]._pGFXLoad = 0;
-		SetPlrAnims(v2);
+
+	if ( myplr == pnum ) {
+		NetSendCmdParam1(TRUE, CMD_PLRDEAD, earflag);//23
+	}
+
+	BOOL diablolevel = gbMaxPlayers > 1 && plr[pnum].plrlevel == 16;
+
+	if ( (DWORD)pnum >= 4 ) {
+		TermMsg("StartPlayerKill: illegal player %d", pnum);
+	}
+
+	if ( plr[pnum]._pClass == PC_WARRIOR ) {
+		PlaySfxLoc(PS_DEAD, plr[pnum].WorldX, plr[pnum].WorldY); // BUGFIX: should use `PS_WARR71` like other classes
+	} else if ( plr[pnum]._pClass == PC_ROGUE ) {
+		PlaySfxLoc(PS_ROGUE71, plr[pnum].WorldX, plr[pnum].WorldY);
+	} else if ( plr[pnum]._pClass == PC_SORCERER ) {
+		PlaySfxLoc(PS_MAGE71, plr[pnum].WorldX, plr[pnum].WorldY);
+	}
+
+	if ( plr[pnum]._pgfxnum ) {
+		plr[pnum]._pgfxnum = 0;
+		plr[pnum]._pGFXLoad = 0;
+		SetPlrAnims(pnum);
 	}
+
 	if ( !(plr[pnum]._pGFXLoad & PFILE_DEATH) ) {
 		LoadPlrGFX(pnum, PFILE_DEATH);
 	}
-	v6 = plr[v3 / 0x54D8]._pDWidth;
-	NewPlrAnim(v2, plr[0]._pDAnim[plr[v3 / 0x54D8]._pdir + v3 / 4], plr[v3 / 0x54D8]._pDFrames, 1, v6);
-	plr[v3 / 0x54D8]._pBlockFlag = 0;
-	plr[v3 / 0x54D8]._pmode = PM_DEATH;
-	plr[v3 / 0x54D8]._pInvincible = 1;
-	SetPlayerHitPoints(v2, 0);
-	v7 = v2 == myplr;
-	plr[v3 / 0x54D8]._pVar8 = 1;
-	if ( !v7 && !itm && !v20 )
-	{
-		v8 = &plr[v3 / 0x54D8].InvBody[0]._itype;
-		v9 = 7;
-		do
-		{
-			*v8 = -1;
-			v8 += 92;
-			--v9;
+
+	PlayerStruct *p = &plr[pnum];
+	NewPlrAnim(pnum, p->_pDAnim[plr[pnum]._pdir], p->_pDFrames, 1, p->_pDWidth);
+
+	plr[pnum]._pBlockFlag = FALSE;
+	plr[pnum]._pmode = PM_DEATH;
+	plr[pnum]._pInvincible = TRUE;
+	SetPlayerHitPoints(pnum, 0);
+	plr[pnum]._pVar8 = 1;
+
+	int i;
+	if ( pnum != myplr && !earflag && !diablolevel ) {
+		for ( i = 0; i < NUM_INVLOC; i++ ) {
+			plr[pnum].InvBody[i]._itype = ITYPE_NONE;
 		}
-		while ( v9 );
-		CalcPlrInv(v2, 0);
+		CalcPlrInv(pnum, 0);
 	}
-	if ( plr[v3 / 0x54D8].plrlevel == currlevel )
-	{
-		FixPlayerLocation(v2, plr[v3 / 0x54D8]._pdir);
-		RemovePlrFromMap(v2);
-		v10 = &dFlags[plr[v3 / 0x54D8].WorldX][plr[v3 / 0x54D8].WorldY];
-		*v10 |= 4u;
-		SetPlayerOld(v2);
-		if ( v2 == myplr )
-		{
+
+	if ( plr[pnum].plrlevel == currlevel ) {
+		FixPlayerLocation(pnum, plr[pnum]._pdir);
+		RemovePlrFromMap(pnum);
+		dFlags[plr[pnum].WorldX][plr[pnum].WorldY] |= 4;
+		SetPlayerOld(pnum);
+
+		if ( pnum == myplr ) {
 			drawhpflag = 1;

```

Comment:
`drawhpflag` should be TRUE instead of 1. This is probably done already with your previous PR.

Reply:
Yeah the other PR does this, I migth have to resolve things when it gets merged...